app/routes/chatbot.js

```diff
@@ -112,74 +114,93 @@ router.post('/', (req, res) => {
       });
   });
 
-  const agentViewOip = process.env.MOBILECOMMONS_OIP_AGENTVIEW;
-
-  return loadUser
-    .then((user) => {
-      logger.info(`loaded user:${user._id}`);
-      scope.user = user;
-
-      let campaign;
-      let campaignId;
-
-      if (scope.broadcast_id) {
-        campaignId = process.env.CAMPAIGNBOT_BROADCAST_CAMPAIGN;
-        campaign = app.locals.campaigns[campaignId];
-
-        if (!campaign) {
-          const msg = `Broadcast Campaign '${campaignId}' not found.`;
-          throw new NotFoundError(msg);
-        }
-
-        // TODO: Add check on app start to trigger alert if Broadcast Campaign is closed.
-        if (campaign.status === 'closed') {
-          // TODO: Include this message to the CampaignClosedError.
-          const msg = `Broadcast Campaign ${campaignId} is closed.`;
-          throw new UnprocessibleEntityError(msg);
+  const loadCampaign = new Promise((resolve, reject) => {
+    logger.log('loadCampaign');
+    let campaign;
+    let campaignId;
+
+    return loadUser
+      .then((user) => {
+        logger.info(`loaded user:${user._id}`);
+        scope.user = user;
+
+        if (scope.broadcast_id) {
+          return app.locals.db.broadcasts
+            .findById(scope.broadcast_id)
+            .then((broadcast) => {
+              if (!broadcast) {
+                // TODO: If this happens, store user/broadcast somewhere to recover lost opportunity
+                // to respond.
+                const err = new Error('broadcast undefined');
+                return reject(err);
+              }
+
+              logger.info(`loaded broadcast:${broadcast._id}`);
+              campaignId = broadcast.campaign;
+              campaign = app.locals.campaigns[campaignId];
+              if (!campaign) {
+                const err = new Error('broadcast campaign undefined');
+                return reject(err);
+              }
+
+              logger.info(`loaded campaign:${campaign._id}`);
+
+              if (campaign.status === 'closed') {
```

Comment:
I'd prefer to keep this as is until we cross that bridge. If we had statuses like `featured` and `active`, `!==active` would not work.